File: WChat/app/model/Message.js
Ext.define('WChat.model.Message', {
    extend: 'Ext.data.Model',

    requires: [
        'Ext.data.Field',
        'Ext.data.proxy.Rest',
        'Ext.DateExtras'
    ],

    config: {
        fields: [
			'messageId',
			'text',
            {
                name: 'timestamp',
                type: 'date',
                convert: function (value, record) {
                    if (value) {
                        return Ext.Date.format(new Date(value), 'o-m-d')
                        
                    } else {
                        return Ext.Date.format(new Date(), 'o-m-d')
                    }
                }
            },
			'userId',
            'roomId'
        ],
        proxy: {
            type: 'rest',
            url: '../api/messages'
        }
    }
});

File: WChatApi/Models/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WChatApi.Models
{
    public class Message
    {
        public Message()
        {
            Timestamp = DateTime.Now;
        }

        public Guid MessageId { get; set; }

        public string Text { get; set; }

        public DateTime Timestamp { get; set; }

        public Guid UserId { get; set; }

        public Guid RoomId { get; set; }
    }
}